tests/manage/mcg/test_bucket_policy.py

```diff
@@ -311,8 +313,414 @@ def test_anonymous_read_only(self, mcg_obj, bucket_factory):
+
+    @pytest.mark.polarion_id("OCS-2140")
+    @tier2
+    def test_bucket_website_and_policies(self, mcg_obj, bucket_factory):
+        """
+        Tests bucket website bucket policy actions
+        """
+        # Creating a OBC (account)
+        obc = bucket_factory(amount=1, interface='OC')
+        obc_obj = OBC(mcg_obj, obc=obc[0].name)
+
+        # Admin sets policy with Put/Get bucket website actions
+        bucket_policy_generated = gen_bucket_policy(
+            user_list=obc_obj.obc_account,
+            actions_list=bucket_website_action_list,
+            resources_list=[obc_obj.bucket_name, f'{obc_obj.bucket_name}/{"*"}'],
+            effect="Allow"
+        )
+        bucket_policy = json.dumps(bucket_policy_generated)
+
+        logger.info(f'Creating bucket policy on bucket: {obc_obj.bucket_name} with principal: {obc_obj.obc_account}')
+        assert helpers.put_bucket_policy(mcg_obj, obc_obj.bucket_name, bucket_policy), "Failed: PutBucketPolicy"
+
+        # Getting Policy
+        logger.info(f'Getting bucket policy for bucket: {obc_obj.bucket_name}')
+        get_policy = helpers.get_bucket_policy(mcg_obj, obc_obj.bucket_name)
+        logger.info('Got bucket policy:%s' % get_policy['Policy'])
+
+        logger.info(f"Adding bucket website config to: {obc_obj.bucket_name}")
+        assert helpers.s3_put_bucket_website(
+            s3_obj=obc_obj, bucketname=obc_obj.bucket_name,
+            website_config=website_config
+        ), "Failed: PutBucketWebsite"
+        logger.info(f"Getting bucket website config from: {obc_obj.bucket_name}")
+        assert helpers.s3_get_bucket_website(s3_obj=obc_obj, bucketname=obc_obj.bucket_name), "Failed: GetBucketWebsite"
+
+        logger.info("Writing index and error data to the bucket")
+        assert helpers.s3_put_object(
+            s3_obj=obc_obj, bucketname=obc_obj.bucket_name,
+            object_key="index.html", data=index, content_type='text/html'
+        ), "Failed: PutObject"
+        assert helpers.s3_put_object(
+            s3_obj=obc_obj, bucketname=obc_obj.bucket_name,
+            object_key="error.html", data=error, content_type='text/html'
+        ), "Failed: PutObject"
+
+        # Verifying whether DeleteBucketWebsite action is denied access
+        logger.info(f'Verifying whether user: {obc_obj.obc_account} is denied to DeleteBucketWebsite')
+        try:
+            helpers.s3_delete_bucket_website(s3_obj=obc_obj, bucketname=obc_obj.bucket_name)
+        except boto3exception.ClientError as e:
+            logger.info(e.response)
+            response = HttpResponseParser(e.response)
+            if response.error['Code'] == 'AccessDenied':
+                logger.info('GetObject action has been denied access')
+            else:
+                raise UnexpectedBehaviour(f"Response received invalid error code {response.error['Code']}")
```

Comment:
not a blocker for merging this PR but may be worth adding the e.response to this exception?

Reply:
ack